src/sentry/replays/scripts/delete_replays.py

```diff
@@ -59,7 +64,42 @@ def translate_cli_tags_param_to_snuba_tag_param(tags: list[str]) -> Sequence[Sea
 
 def delete_replay_ids(project_id: int, replay_ids: list[str]) -> None:
     """Delete a set of replay-ids for a specific project."""
+    logger.info("Archiving %d replays.", len(replay_ids))
+
+    # Bulk produce archived replay rows to the ingest-replay-events topic before flushing.
+    #
+    # This operation is fast enough that it can be performed synchronously. Archiving
+    # synchronously gives the script runner the feeling that the script executed to completion
+    # and will allow them to immediately spot-check before moving on with their day. In a
+    # purely asynchronous world the script runner will have to continually worry if their tasks
+    # executed.
+    #
+    # This also gives us reasonable assurances that if the script ran to completion the customer
+    # will not be able to access their deleted data even if the actual deletion takes place some
+    # time later.
+    with _bulk_produce_then_flush() as publisher:
+        for replay_id in replay_ids:
+            archive_replay(publisher, project_id, replay_id)
+
+    logger.info("Scheduling %d replays for deletion.", len(replay_ids))
+
+    # Asynchronously delete RRWeb recording data.
+    #
+    # Because this operation could involve millions of requests to the blob storage provider we
+    # schedule the tasks to run on a cluster of workers. This allows us to parallelize the work
+    # and complete the task as quickly as possible.
     for replay_id in replay_ids:
-        delete_replay_recording(project_id, replay_id)
+        delete_recording_segments.delay(project_id, replay_id)
```

Comment:
this task already calls archive, can make delete_replay_recording a task and call that instead